slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -105,20 +103,19 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
           )
         ).mkString(", ")
         if(classEnabled){
-          val prns = (parents.take(1).map(" extends "+_) ++ parents.drop(1).map(" with "+_)).mkString("")
-          val companion: String = if(columns.size >= 22) s"""
+          val _parents = (parents.take(1).map(" extends "+_) ++ parents.drop(1).map(" with "+_)).mkString("")
+          val companion: String =
+            if(columns.size >= 22)(
+              s"""
 object $name{
-  def unapply(x: $name ) = {
-    val data = ${compoundValue(columns.map(_.name) , "x.")}
-    Option[data.type](data)
+  def unapply(x: $name) = {
+    val data = ${compoundValue(columns.map(_.name), "x.")}
+    Option(data)
   }
 }
-"""
-                                  else ""
-s"""
-$companion
-case class $name($args)$prns
-"""
+              """.trim + "\n"
+            ) else ""
+          s"""${companion}case class $name($args)${_parents}"""
```

Comment:
the above introduced unnecessary new newlines into the generated code that made it look weird. let's avoid that. I try to make generated code as nicely readable as possible